hibernate-spatial/src/test/java/org/hibernate/spatial/testing/RequireFunctionExtension.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.spatial.testing;
+
+import java.util.Locale;
+import java.util.Optional;
+
+import org.hibernate.query.sqm.function.SqmFunctionRegistry;
+import org.hibernate.spatial.contributor.SpatialFunctionContributor;
+
+import org.hibernate.testing.orm.junit.JUnitHelper;
+import org.hibernate.testing.orm.junit.SessionFactoryExtension;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.TestingUtil;
+import org.junit.jupiter.api.extension.ConditionEvaluationResult;
+import org.junit.jupiter.api.extension.ExecutionCondition;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import org.jboss.logging.Logger;
+
+public class RequireFunctionExtension implements ExecutionCondition {
+
+	private static final ConditionEvaluationResult ENABLED = ConditionEvaluationResult.enabled(
+			"No applicable @RequireFunction annotation" );
+
+	private static final Logger log = Logger.getLogger( RequireFunctionExtension.class );
+
+	@Override
+	public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext context) {
+		Optional<Object> testInstance = context.getTestInstance();
+
+		if ( !testInstance.isPresent() ) {
+			return ENABLED;
+		}
+
+		final SqmFunctionRegistry functionRegistry = new SqmFunctionRegistry();
+
+		ExtensionContext.Store store = JUnitHelper.locateExtensionStore(
+				SessionFactoryExtension.class,
+				context,
+				testInstance.get()
+		);
+
+		final SessionFactoryScope existing = (SessionFactoryScope) store.get( SessionFactoryScope.class.getName() );
+
+		if ( existing == null ) {
+			return ConditionEvaluationResult.enabled( "" );
+		}
+
+		final Optional<RequiresFunction> requiresFunctions = TestingUtil.findEffectiveAnnotation(
+				context,
+				RequiresFunction.class
+		);
+
+		if ( requiresFunctions.isPresent() ) {
+			String functionKey = requiresFunctions.get().key();
+			SpatialFunctionContributor contributor = new SpatialFunctionContributor();
+			contributor.contributeFunctions(
+					functionRegistry,
+					existing.getSessionFactory().getServiceRegistry()
+			);
```

Comment:
I'm a bit surprised that this is needed. Isn't the service loader contract supposed to do the registrations?

Reply:
The functions are registered here in a function registry, separate from the QueryEngine's registry, so we can disable tests for dialects where the function under tests is not supported. If I remember correctly, I needed this before the Session is built and the service loader comes into play.

Reply:
Do you remember the reason for this? Would be great to add a comment in the code with the reason for this. We had a few issues related to DDL that required the introduction of `DialectContext` to check for dialect features before the session factory is built. Since you are referring to a session factory here, I don't think the reason can be a DDL problem, as that has run by the time this code runs.

Reply:
I had a look at it again, and you were right. I somehow missed that I could access from the `SessionFactoryImplementor` the SqmFunctionRegistry, and it has by then all the functions. I will remove this code.